docs/utilities/data_masking.md

```diff
@@ -97,15 +97,19 @@ Before you start, you will need a KMS key to encrypt and decrypt your data. Your
 
 ### Masking data
 
-You can mask data without having to install any encryption library. Masking data will result in the loss of its original type, and the masked data will always be represented as a string.
+!!! note "You can mask data without [installing any dependency](#install)."
+
+Masking will erase the original data and replace with `*****`. This means you cannot recover masked data, and its type will change to `str`.
```

Comment:
Should we add here that the type will change to `str` unless the data to be masked is of an Iterable type like `list`, `tuple`, or `set`, in which case it will return a new object of the same type as the input data but with each element replaced by the string `*****`?

Reply:
Great catch! I was gonna add that to the `Fields` piece, but you're 100% right.. one might not reach that section and take it for granted.

We need to encapsulate that into the static typing return too.

Reply:
<img width="750" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/3340292/ff01c01f-9a53-41af-8f5b-e704c34a7edd">